File: scrolltotop/src/main/java/com/mahdikh/vision/scrolltotop/animator/ScaleAnimator.java
package com.mahdikh.vision.scrolltotop.animator;

import android.view.View;

public class ScaleAnimator extends FadeAnimator {
    private float fromScale;

    public ScaleAnimator(float maxAlpha, float fromScale) {
        super(maxAlpha);
        this.fromScale = fromScale;
    }

    public ScaleAnimator(float fromScale) {
        this.fromScale = fromScale;
    }

    public ScaleAnimator() {
        fromScale = 0.85F;
    }

    @Override
    public void onPrepare(View view) {
        super.onPrepare(view);
        view.setScaleX(fromScale);
        view.setScaleY(fromScale);
    }

    @Override
    public void onShow(View view) {
        view.setScaleY(fromScale);
        view.setScaleX(fromScale);

        super.onShow(view);

        view.animate()
                .scaleX(1.0F)
                .scaleY(1.0F);
    }

    @Override
    public void onHide(View view) {
        super.onHide(view);
        view.animate()
                .scaleX(fromScale)
                .scaleY(fromScale);
    }

    public ScaleAnimator setFromScale(float fromScale) {
        this.fromScale = fromScale;
        return this;
    }
}

File: scrolltotop/src/main/java/com/mahdikh/vision/scrolltotop/animator/ScaleAnimator.kt
package com.mahdikh.vision.scrolltotop.animator

import android.view.View

open class ScaleAnimator : FadeAnimator{
    var fromScale: Float

    constructor(maxAlpha: Float, fromScale: Float) : super(maxAlpha) {
        this.fromScale = fromScale
    }

    constructor(fromScale: Float) {
        this.fromScale = fromScale
    }

    constructor() {
        fromScale = 0.85f
    }

    override fun prepare(view: View) {
        super.prepare(view)
        view.apply {
            scaleX = fromScale
            scaleY = fromScale
        }
    }

    override fun preAnimateShow(view: View) {
        super.preAnimateShow(view)
        view.apply {
            scaleX = fromScale
            scaleY = fromScale
        }
    }

    override fun animateShowImpl(view: View) {
        super.animateShowImpl(view)
        view.animate().apply {
            scaleY(1.0F)
            scaleX(1.0F)
            duration = this@ScaleAnimator.duration
            interpolator  = this@ScaleAnimator.interpolator
        }.start()
    }

    override fun animateHideImpl(view: View) {
        super.animateHideImpl(view)
        view.animate().apply {
            scaleX(fromScale)
            scaleY(fromScale)
            duration = this@ScaleAnimator.duration
            interpolator = this@ScaleAnimator.interpolator
        }.start()
    }
}